src/marshmallow/fields.py

```diff
@@ -651,15 +657,24 @@ def __init__(self, tuple_fields, *args, **kwargs):
             )
 
         self.validate_length = Length(equal=len(self.tuple_fields))
+        for container in self.tuple_fields:
```

Comment:
I'm not sure it makes sense for `Tuple` to propagate `only` and `exclude`. 

The code here introduces some unexpected behavior.

For example:

```python
from marshmallow import Schema, fields


class Child(Schema):
    name = fields.String()
    age = fields.Integer()


class Family(Schema):
    children = fields.Tuple((fields.Nested(Child, only=("age",)), fields.Nested(Child)))


schema = Family()

print(
    schema.dump(
        {"children": ({"name": "steve", "age": 0}, {"name": "jerome", "age": 0})}
    )
)
```

Expected:

```
{'children': ({'age': 0}, {'name': 'jerome', 'age': 0})}
```

Actual:

```
{'children': ({'age': 0}, {'age': 0})}
```

Reply:
Oh, right. I didn't see that.

I'm still unhappy with the whole propagation mechanism, but I'm not sure we can do much better without a larger rework (https://github.com/marshmallow-code/marshmallow/issues/1244).